plugins/reach/lib/kReachManager.php

```diff
@@ -780,60 +772,103 @@ public static function addEntryVendorTask(entry $entry, ReachProfile $reachProfi
 		$entryVendorTask->setTurnAroundTime($vendorCatalogItem->getTurnAroundTime());
 
 		if ($context)
+		{
 			$entryVendorTask->setContext($context);
+		}
 
 		if ($creationMode)
+		{
 			$entryVendorTask->setCreationMode($creationMode);
+		}
 
-		$status = EntryVendorTaskStatus::PENDING;
 		if ($validateModeration && $reachProfile->shouldModerate($vendorCatalogItem->getServiceType()))
 		{
 			$entryVendorTask->setIsRequestModerated(true);
-			$status = EntryVendorTaskStatus::PENDING_MODERATION;
 		}
-		
-		if($vendorCatalogItem->requiresEntryReady() && $entry->getStatus() != entryStatus::READY)
+
+		$status = self::getEntryVendorTaskStatus($reachProfile, $vendorCatalogItem, $entry, $entryObjectType, $validateModeration);
+		$entryVendorTask->setStatus($status);
+
+		$dictionary = $reachProfile->getDictionaryByLanguage($vendorCatalogItem->getSourceLanguage());
+		if ($dictionary)
 		{
-			$status = EntryVendorTaskStatus::PENDING_ENTRY_READY;
+			$entryVendorTask->setDictionary($dictionary->getData());
 		}
-		
-		//Kaltura Recorded entries are ready on creation so make sure the vendors wont fetch the job until it gets its assets
-		if($entry->getSourceType() == EntrySourceType::KALTURA_RECORDED_LIVE && $vendorCatalogItem->requiresEntryReady())
+
+		return $entryVendorTask;
+	}
+
+	protected static function shouldAddEntryVendorTask($entry, $entryObjectType, $vendorCatalogItem)
+	{
+		if($entryObjectType == KalturaEntryObjectType::ENTRY)
 		{
-			$entryAssets = assetPeer::retrieveReadyFlavorsByEntryId($entry->getId());
-			if(!count($entryAssets))
+			//Check if the entry is temporary, if so, dont create the task
+			if($entry->getIsTemporary())
+			{
+				KalturaLog::debug("Entry [{$entry->getId()}] is temporary, entry vendor task object wont be created for it");
+				return false;
+			}
+
+			//Check if static content and the catalog item is excluding static content, if so, dont create the task
+			if(count($vendorCatalogItem->getAdminTagsToExcludeArray()) && array_intersect($vendorCatalogItem->getAdminTagsToExcludeArray(), $entry->getAdminTagsArr()))
+			{
+				KalturaLog::debug("Entry [{$entry->getId()}] has admin tags that are excluded by the catalog item, entry vendor task object wont be created for it");
+				return false;
+			}
```

Comment:
in case $entryObjectType == KalturaEntryObjectType::ENTRY we should allow, but in other object types we should deny (until new object type is added to the code)